core/src/main/scala/scala/pickling/generator/sourcegen.scala

```diff
@@ -400,6 +396,11 @@ private[pickling]  trait SourceGenerator extends Macro with tags.FastTypeTagMacr
         implicit object $picklerName extends _root_.scala.pickling.Pickler[$tpe] with _root_.scala.pickling.Generated {
           def pickle(picklee: $tpe, builder: _root_.scala.pickling.PBuilder): _root_.scala.Unit = ${genPicklerLogic[T](picklerAst)}
           val tag: _root_.scala.pickling.FastTypeTag[$tpe] = $createTagTree
+
+          if(_root_.scala.pickling.internal.currentRuntime.picklers.lookupPickler(tag.key).isEmpty) {
```

Comment:
There's a problem here, though. Both pickler/unpickler are implicit objects and if they get referenced they will be instantiated. So, I guess there's no way to be more aggresive, since in order to check that a pickler/unpickler is registered we need to access one field of the object.
  
Regarding your second suggestion, should I then only generate both picklers and unpicklers? I think it's a great idea and that's actually what we do in spores-pickling, but I'll do that in an independent PR. WDYT?
  
Also, if you wait for 5 min, I'm submitting some changes to this PR.